src/compiler/compile/render_dom/wrappers/Element/index.ts

```diff
@@ -1015,6 +1015,15 @@ function to_html(wrappers: Array<ElementWrapper | TextWrapper | MustacheTagWrapp
 			if (!wrapper.void) {
 				state.quasi.value.raw += '>';
 
+				if (wrapper.node.name === 'pre') {
+					// Two or more leading newlines are required to restore the leading newline immediately after `<pre>`.
+					// see https://html.spec.whatwg.org/multipage/grouping-content.html#the-pre-element
```

Comment:
from https://html.spec.whatwg.org/multipage/syntax.html#element-restrictions

i think we should add the same logic for `<textarea>` as well

Reply:
I didn't know that. Thanks for letting me know.